src/GraphQL.php

```diff
@@ -118,6 +119,17 @@ public function schema($schema = null): Schema
      */
     public function query($query, ?array $variables = [], array $opts = []): array
     {
+        $middlewareResponse = app()->make(Pipeline::class)
+            ->send([$query, $variables])
+            ->through($this->executionMiddleware())
+            ->via('resolve')
+            ->thenReturn();
+
+        if (1 === count($middlewareResponse)) {
+            return $middlewareResponse;
+        }
```

Comment:
this catches, if you returning directly in the middleware.

After thinking about it this might not be enough to know for sure that it was a response that has been returned, will investigate further